### PR TITLE
Car added

### PR DESCRIPTION
**Pull request by plitvica**
properties, getter & setter added

### CODE REVIEW

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 1]
```diff
@@ -2,5 +2,63 @@
 
 import at.fhv.itm3.s2.roundabout.api.entity.ICar;
 
+import java.time.LocalTime;
+import java.util.List;
+
 public class Car implements ICar {
+
+    private LocalTime lastUpdateTime;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: why LocalTime? i know its from the entity diagram, but isn't it the simulation time (integer or long)?

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 2]
```diff
@@ -2,5 +2,63 @@
 
 import at.fhv.itm3.s2.roundabout.api.entity.ICar;
 
+import java.time.LocalTime;
+import java.util.List;
+
 public class Car implements ICar {
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: constructor missing, for example to initialize the route list (or set the actual section)

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 3]
```diff
@@ -2,5 +2,63 @@
...
+
+    private LocalTime lastUpdateTime;
+    private DriverBehaviour driverBehaviour;
+    private double length;
+    private StreetSection streetSection;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: destination, not streetSection

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 4]
```diff
@@ -2,5 +2,63 @@
...
+    private List<StreetSection> route;
+    private StreetSection actualSection;
+
+    public LocalTime getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    public void setLastUpdateTime(LocalTime lastUpdateTime) {
+        this.lastUpdateTime = lastUpdateTime;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: checks, for example if positive

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 5]
```diff
@@ -2,5 +2,63 @@
...
+    public void setDriverBehaviour(DriverBehaviour driverBehaviour) {
+        this.driverBehaviour = driverBehaviour;
+    }
+
+    public double getLength() {
+        return length;
+    }
+
+    public void setLength(double length) {
+        this.length = length;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: checks, for example if positive

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 6]
```diff
@@ -2,5 +2,63 @@
...
+    public void setRoute(List<StreetSection> route) {
+        this.route = route;
+    }
+
+    public StreetSection getActualSection() {
+        return actualSection;
+    }
+
+    public void setActualSection(StreetSection actualSection) {
+        this.actualSection = actualSection;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: checks, for example if it is in route list and "newer" than actual section

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java (hunk 1 above)
Comment (plitvica) [reply roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: jep, that was the reason, i think it was double

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 7]
```diff
@@ -1,6 +1,81 @@
...
+    private DriverBehaviour driverBehaviour;
+    private double length;
+    private StreetSection destination;
+    private List<StreetSection> route;
+    private StreetSection actualSection;
+
+    public Car(){}
+
+    public Car(long lastUpdateTime, DriverBehaviour driverBehaviour, double length, StreetSection destination, List<StreetSection> route, StreetSection actualSection ){
+        this.lastUpdateTime = lastUpdateTime;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: imho this is 0 when the object is created

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 8]
```diff
@@ -1,6 +1,81 @@
...
+    }
+
+    public double getLength() {
+        return length;
+    }
+
+    public void setLength(double length) {
+        if(length > 0) {
+            this.length = length;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: else throw exception

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 9]
```diff
@@ -1,6 +1,81 @@
...
+    }
+
+    public double getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    public void setLastUpdateTime(double lastUpdateTime) {
+        if(lastUpdateTime > 0){
+            this.lastUpdateTime = lastUpdateTime;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: else throw exception

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java [hunk 10]
```diff
@@ -1,6 +1,81 @@
...
+    }
+
+    public StreetSection getActualSection() {
+        return actualSection;
+    }
+
+    public void setActualSection(StreetSection actualSection) {
+        if(route.contains(actualSection) && route.indexOf(actualSection) >= route.indexOf(this.actualSection)){
+            this.actualSection = actualSection;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java]: else throw exception

**Pull request merged**